x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/use_alert_suppression.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback } from 'react';
+import { isSuppressibleAlertRule } from '../../../../../common/detection_engine/utils';
+import { SuppressibleAlertRules } from '../../../../../common/detection_engine/constants';
+import type { ExperimentalFeatures } from '../../../../../common';
+import { useIsExperimentalFeatureEnabled } from '../../../../common/hooks/use_experimental_features';
+import type { RuleResponse } from '../../../../../common/api/detection_engine/model/rule_schema';
+
+export interface UseAlertSuppressionReturn {
+  isSuppressionEnabled: boolean;
+}
+
+export const useAlertSuppression = (rule: Partial<RuleResponse>): UseAlertSuppressionReturn => {
+  const IsRuleFeatureFlagEnabled = (ruleFFName: keyof ExperimentalFeatures) =>
```

Comment:
```suggestion
  const isRuleFeatureFlagEnabled = (ruleFFName: keyof ExperimentalFeatures) =>
```

usually variables that starts with capital letter are types or classes. To avoid any confusion, would suggest to rename it with lowercase `i`

Reply:
Also, React Components otherwise I will get the below error

`React Hook "useIsExperimentalFeatureEnabled" is called in function "isRuleFeatureFlagEnabled" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use"`

Reply:
Actually, I discovered an issue here since this method is not correct because it will lead to calling the `useIsExperimentalFeatureEnabled` hook conditionally which is not supported by React as it collects all the hooks/states in an array and this won't make the `useIsExperimentalFeatureEnabled` registered in the first run so that I removed at all!